File: src/main/resources/generators/scala/native/DateWrapper.scala
import java.util.Date

// TODO: Use newer Date API
class DateWrapper(private val inner: Date) {
  def timestamp: NumberWrapper = new NumberWrapper(inner.getTime())

  def day: NumberWrapper = new NumberWrapper(inner.getDate)

  def month: NumberWrapper = new NumberWrapper(inner.getMonth)

  def ==(dateWrapper: DateWrapper): BooleanWrapper = timestamp == dateWrapper.timestamp

  def !=(dateWrapper: DateWrapper): BooleanWrapper = timestamp != dateWrapper.timestamp

  def >(dateWrapper: DateWrapper): BooleanWrapper = timestamp > dateWrapper.timestamp

  def <(dateWrapper: DateWrapper): BooleanWrapper = timestamp < dateWrapper.timestamp

  def >=(dateWrapper: DateWrapper): BooleanWrapper = timestamp >= dateWrapper.timestamp

  def <=(dateWrapper: DateWrapper): BooleanWrapper = timestamp <= dateWrapper.timestamp
}

File: src/main/resources/generators/ts/native/DateWrapper.ts
export class DateWrapper {
    private inner: Date;

    constructor(inner: Date) {
        this.inner = inner;
    }

    get timestamp() {
        return new NumberWrapper(this.inner.getTime());
    }

    get day() {
        return new NumberWrapper(this.inner.getDate());
    }

    get month() {
        return new NumberWrapper(this.inner.getMonth());
    }

    toDate() {
        return this.inner;
    }

    equals(date: DateWrapper): boolean {
        return this.timestamp == date.timestamp;
    }

    notEquals(date: DateWrapper): boolean {
        return this.timestamp != date.timestamp;
    }

    upper(date: DateWrapper): boolean {
        return this.timestamp > date.timestamp;
    }

    lower(date: DateWrapper): boolean {
        return this.timestamp < date.timestamp;
    }

    upperOrEquals(date: DateWrapper): boolean {
        return this.timestamp >= date.timestamp;
    }

    lowerOrEquals(date: DateWrapper): boolean {
        return this.timestamp <= date.timestamp;
    }
}